src/components/AmplitudeAnalytics/index.ts

```diff
@@ -0,0 +1,109 @@
+      trackingOptions: {
+        ipAddress: false,
+        carrier: false,
+        city: false,
+        region: false,
+        country: false,
+        dma: false, // Disables designated market area tracking
+      },
+    }
+  )
+}
+
+/** Sends an event to Amplitude. */
+export function sendAnalyticsEvent(eventName: string, eventProperties?: Record<string, unknown>) {
+  if (process.env.NODE_ENV === 'development') {
+    console.log('amplitude event log:', `${eventName}: ${JSON.stringify(eventProperties)}`)
+    return
+  }
+
+  track(eventName, eventProperties)
+}
+
+/**
+ * Class that exposes methods to modify the User Model's properties in
+ * Amplitude that represents the current session's user.
+ *
+ * See https://help.amplitude.com/hc/en-us/articles/115002380567-User-properties-and-event-properties
+ * for details.
+ */
+class UserModel {
+  private _isDevEnvironemnt = true
+
+  constructor() {
+    process.env.NODE_ENV === 'development' ? (this._isDevEnvironemnt = true) : (this._isDevEnvironemnt = false)
+  }
+
+  public set(propertyName: string, propertyValue: string | number) {
+    if (this._isDevEnvironemnt) {
+      console.log('amplitude user model update with operation set:', `${propertyName}: ${propertyValue}`)
+      return
+    }
+    const identifyObj = new Identify()
+    identifyObj.set(propertyName, propertyValue)
+    identify(identifyObj)
+  }
+
+  public setOnce(propertyName: string, propertyValue: string | number) {
+    if (this._isDevEnvironemnt) {
+      console.log('amplitude user model update with operation setOnce:', `${propertyName}: ${propertyValue}`)
+      return
+    }
+    const identifyObj = new Identify()
+    identifyObj.setOnce(propertyName, propertyValue)
+    identify(identifyObj)
+  }
+
+  public add(propertyName: string, propertyValue: string | number) {
+    if (this._isDevEnvironemnt) {
+      console.log('amplitude user model update with operation add:', `${propertyName}: ${propertyValue}`)
+      return
+    }
+    const identifyObj = new Identify()
+    identifyObj.add(propertyName, typeof propertyValue === 'number' ? propertyValue : 0)
+    identify(identifyObj)
+  }
+
+  public postInsert(propertyName: string, propertyValue: string | number) {
+    if (this._isDevEnvironemnt) {
+      console.log('amplitude user model update with operation postInsert:', `${propertyName}: ${propertyValue}`)
+      return
+    }
+    const identifyObj = new Identify()
+    identifyObj.postInsert(propertyName, propertyValue)
+    identify(identifyObj)
+  }
+
+  public remove(propertyName: string, propertyValue: string | number) {
+    if (this._isDevEnvironemnt) {
+      console.log('amplitude user model update with operation remove:', `${propertyName}: ${propertyValue}`)
+      return
+    }
+    const identifyObj = new Identify()
+    identifyObj.remove(propertyName, propertyValue)
+    identify(identifyObj)
+  }
+}
+
+export const userModel = new UserModel()
```

Comment:
my gut says this should be a capital case so how about changing the class to `_UserModel` and then you can `export const UserModel`. also i'm not tied to the name user model if you can think of something better.